test/model/message_list_test.dart

```diff
@@ -271,6 +271,68 @@ void main() {
     checkInvariants(model);
   });
 
+  group('DeleteMessageEvent', () {
+    final stream = eg.stream();
+    final messages = List.generate(30, (i) => eg.streamMessage(stream: stream));
+
+    test('in narrow', () async {
+      await prepare(narrow: StreamNarrow(stream.streamId));
+      await prepareMessages(foundOldest: true, messages: messages);
+
+      check(model).messages.length.equals(30);
+      await store.handleEvent(eg.deleteMessageEvent(messages.sublist(0, 10)));
+      checkNotifiedOnce();
+      check(model).messages.length.equals(20);
+    });
+
+    test('not all in narrow', () async {
+      await prepare(narrow: StreamNarrow(stream.streamId));
+      await prepareMessages(foundOldest: true, messages: messages.sublist(5));
+
+      check(model).messages.length.equals(25);
+      await store.handleEvent(eg.deleteMessageEvent(messages.sublist(0, 10)));
+      checkNotifiedOnce();
+      check(model).messages.length.equals(20);
+    });
+
+    test('not in narrow', () async {
+      await prepare(narrow: StreamNarrow(stream.streamId));
+      await prepareMessages(foundOldest: true, messages: messages.sublist(5));
+
+      check(model).messages.length.equals(25);
+      await store.handleEvent(eg.deleteMessageEvent(messages.sublist(0, 5)));
+      checkNotNotified();
+      check(model).messages.length.equals(25);
+      checkInvariants(model);
+    });
+
+    test('complete message deletion', () async {
+      await prepare(narrow: StreamNarrow(stream.streamId));
+      await prepareMessages(foundOldest: true, messages: messages.sublist(0, 25));
+
+      check(model).messages.length.equals(25);
+      await store.handleEvent(eg.deleteMessageEvent(messages));
+      checkNotifiedOnce();
+      check(model).messages.length.equals(0);
+    });
+
+    test('non-consecutive message deletion', () async {
+      await prepare(narrow: StreamNarrow(stream.streamId));
+      await prepareMessages(foundOldest: true, messages: messages);
+      final messageToDelete = messages.sublist(2, 5) + messages.sublist(10, 15);
+
+      check(model).messages.length.equals(30);
+      await store.handleEvent(eg.deleteMessageEvent(messageToDelete));
+      checkNotifiedOnce();
+      final remainingMessageIds = [
+        ...messages.sublist(0, 2),
+        ...messages.sublist(5, 10),
+        ...messages.sublist(15)].map((message) => message.id).toSet();
+      check(model).messages.length.equals(22);
+      check(model.messages.map((message) => message.id)).containsInOrder(remainingMessageIds);
```

Comment:
Hmm, this doesn't seem like the right check. It checks that all the elements from the expected iterable are present in the actual iterable, and in the same order… but the expected iterable is a Set, which doesn't necessarily iterate through its elements in any predictable order.

And in fact the `Iterable.toSet` method's doc explicitly says:
> The order of the elements in the set is not guaranteed to be the same as for the iterable.

So this test could flake if that set's iteration order changes.

How about leaving off the `toSet`, and using `.deepEquals(remainingMessageIds)`?